js/client-rooms.js

```diff
@@ -19,6 +23,14 @@
 			app.user.on('change:named', this.updateUser, this);
 			this.update();
 		},
+		initSectionSelection: function () {
+			var buf = ['<option value="all">All rooms</option>'];
+			for (var i = 0; i < app.roomsData.sectionTitles.length; i++) {
+				var sectionName = app.roomsData.sectionTitles[i];
+				buf.push('<option value="' + toID(sectionName) + '">' + sectionName + '</option>');
```

Comment:
Say it with me: html~ injection~ vulnerability~

Reply:
(Say it with me: Finishing Preact is a better use of your time than making a bunch of UI changes to the legacy client)